server/app/views/components/CreateQuestionButton.java

```diff
@@ -0,0 +1,64 @@
+package views.components;
+
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.p;
+
+import j2html.tags.specialized.ATag;
+import j2html.tags.specialized.DivTag;
+import services.question.types.QuestionType;
+import views.style.StyleUtils;
+import views.style.Styles;
+
+/**
+ * Contains a helper method to render the create question button with it's corresponding dropdown.
+ */
+public final class CreateQuestionButton {
+  /** Renders the "Create new question" button with a dropdown for each question type. */
+  public static DivTag renderCreateQuestionButton() {
+    String parentId = "create-question-button";
+    String dropdownId = parentId + "-dropdown";
+    DivTag linkButton =
+        new LinkElement().setId(parentId).setText("Create new question").asButtonNoHref();
```

Comment:
This is pre-existing code but still few comments. No need to do anything. But if you have background on any of these - would be good for me to learn.
1. Why do we use link as button here while in other places we use regular <button> element.
2. `asButtonNoHref` implies that this element cast as Button (whatever it means) while in fact it actually wraps in a <div> tag.

Reply:
Good callout. `LinkElement` is a helper that I'm trying to move us away from for some of the reasons you call out. I switched this to just create a button tag with the appropriate common styling. Calling this a link when it's really a button with a dropout is confusing :).